pkg/awsutils/awsutils.go

```diff
@@ -224,8 +225,66 @@ func prometheusRegister() {
 	}
 }
 
+//StringSet is a set of strings
+type StringSet struct {
+	sync.RWMutex
+	data 	sets.String
+}
+
+func (ss *StringSet) AWSStrings() []*string {
+	ss.RLock()
+	defer ss.RUnlock()
+	var dataSlice []*string
+	for key, _ := range ss.data {
+		dataSlice = append(dataSlice, aws.String(key))
+	}
+	return dataSlice
+}
+
+func (ss *StringSet) Set(items []string) {
+	ss.Lock()
+	defer ss.Unlock()
+	for _, item := range items {
+		ss.data.Insert(item)
+	}
+}
+
+func (ss *StringSet) IsEmpty() bool {
+	if ss.data.Len() == 0 {
+		ss.data = sets.NewString()
+		return true
+	}
+	return false
+}
+
+func (ss *StringSet) compareAddDel (items []string) {
+	newSet := StringSet{data: sets.String{}}
+	newSet.Set(items)
+	//For example: s1 = {a1, a2, a3} s2 = {a1, a2, a4, a5} s1.Difference(s2) = {a3} s2.Difference(s1) = {a4, a5}
+	setDiff := ss.data.Difference(newSet.data)
+	if setDiff.Len() != 0 {
+		ss.Lock()
+		defer ss.Unlock()
+		for delItem, _ := range setDiff {
+			ss.data.Delete(delItem)
+			log.Infof("Removed %s from ipamd cache", delItem)
```

Comment:
Maybe we should just call it a set and quote the string?
```go
log.Infof("Removed '%s' from StringSet", delItem)
```
Same for the add log line.

Reply:
No, these log messages should not be in the `StringSet` implementation. See below for where they should be.